options/options.go

```diff
@@ -4,26 +4,63 @@ import (
 	"fmt"
 	"net/url"
 	"reflect"
+	"strings"
 )
 
 func Encode(opts interface{}) string {
+	return extract(opts).Encode()
+}
+
+func extract(opts interface{}) url.Values {
 	v := url.Values{}
 	ot := reflect.TypeOf(opts)
 	ov := reflect.ValueOf(opts)
 
 	for i := 0; i < ot.NumField(); i++ {
-		otf := ot.Field(i)
 		ovf := ov.Field(i)
 
-		if !ovf.IsNil() {
-			p := otf.Tag.Get("param")
-			if p == "" {
-				p = otf.Name
+		if ovf.Kind() == reflect.Invalid {
```

Comment:
please remove this check, it's unecessary because this case will already be handled by the check `ovf.Kind() != reflect.Pointer` line 46 now.
(sorry I had removed this in my suggestion in an edit but maybe it was too late)